src/translate.js

```diff
@@ -87,6 +87,9 @@ function getFunctionFromDefaultAndShaderValue(sceneDrawGroup, ccssProperty, defa
     }
     var fn = `var _value=${defaultValue};`;
     shaderValue.js.forEach(function (code) {
+        if (code.search(/data\['mapnik::\S+'\]/) >= 0) {
```

Comment:
Why data? Also, what about if mapnik::geometry_type is defined within " instead of '?

Reply:
Also, if there is spaces between the [ and the ' | " the RegExp doesn't work either.

Reply:
The line is searching for regexps coincidences in the https://github.com/CartoDB/carto JS generated code. This is a workaound for  the coupling between https://github.com/CartoDB/carto and mapnik. Since the JS code produced by https://github.com/CartoDB/carto doesn't provide more information to avoid this problem this is the best that can be done at this repo.

It would be possible to fix this on https://github.com/CartoDB/carto , but I don't think that its worth the trouble: it will be certainly difficult to fix and, if done incorrectly, it can cause problems with the mapnik renderer.

Reply:
> Also, if there is spaces between the [ and the ' | " the RegExp doesn't work either.

  You're completely right, but that is generated by carto, we could use a less strict regexp or leave it like that knowing that carto is not under very active development right now.

Reply:
I don't want to be the blocker guy, so this is your decision :).